features/nanostack/targets/TARGET_NXP/TARGET_KW41Z/NanostackRfPhyKw41z.cpp

```diff
@@ -0,0 +1,1093 @@
+    ZLL->PHY_CTRL &= ~ZLL_PHY_CTRL_TRCV_MSK_MASK;
+
+    NVIC_ClearPendingIRQ(Radio_1_IRQn);
+    NVIC_EnableIRQ(Radio_1_IRQn);
+}
+
+/*
+ * \brief Function reads a time-stamp value from XCVR [symbols]
+ *
+ * \param pEndTime pointer to location where time-stamp will be stored
+ *
+ * \return none
+ */
+static void rf_get_timestamp(uint32_t *pRetClk)
+{
+    if (NULL == pRetClk) {
+        return;
+    }
+
+    platform_enter_critical();
+
+    *pRetClk = 0;
+    *pRetClk = ZLL->EVENT_TMR >> ZLL_EVENT_TMR_EVENT_TMR_SHIFT;
+
+    platform_exit_critical();
+}
+
+/*
+ * \brief Function starts the ACK wait time-out.
+ *
+ * \param slots The ACK wait time-out in [symbols]
+ *
+ * \return none
+ */
+static void rf_ack_wait_timer_start(uint16_t time)
+{
+    uint32_t timestamp, t;
+
+    rf_get_timestamp(&timestamp);
+
+    t = (rf_get_timeout() - timestamp) & gPhyTimeMask_c;
+
+    if (t > 1) {
+        timestamp += time;
+        rf_set_timeout(timestamp);
+    }
+}
+
+/*
+ * \brief Function sets the RF in RX state.
+ *
+ * \param none
+ *
+ * \return none
+ */
+static void rf_receive(void)
+{
+    uint32_t irqSts;
+
+    /* RX can start only from Idle state */
+    if (mPhySeqState != gIdle_c) {
+        rf_abort();
+    }
+
+    mPhySeqState = gRX_c;
+
+    /* Ensure that no spurious interrupts are raised, but do not change TMR1 and TMR4 IRQ status */
+    irqSts = ZLL->IRQSTS;
+    irqSts &= ~(ZLL_IRQSTS_TMR1IRQ_MASK | ZLL_IRQSTS_TMR4IRQ_MASK);
+    irqSts |= ZLL_IRQSTS_TMR3MSK_MASK;
+    ZLL->IRQSTS = irqSts;
+
+    /* unmask SEQ interrupt */
+    ZLL->PHY_CTRL &= ~ZLL_PHY_CTRL_SEQMSK_MASK;
+
+    /* Start the RX sequence */
+    ZLL->PHY_CTRL |= gRX_c;
+}
+
+static void rf_mac_ed_state_enable(void)
+{
+    uint32_t ccaMode, irqSts;
+
+    mPhySeqState = gCCA_c;
+
+    /* Switch to ED mode */
+    ccaMode = (ZLL->PHY_CTRL  & ZLL_PHY_CTRL_CCATYPE_MASK) >> ZLL_PHY_CTRL_CCATYPE_SHIFT;
+    if (ccaMode != gCcaED_c) {
+        ZLL->PHY_CTRL &= ~(ZLL_PHY_CTRL_CCATYPE_MASK);
+    }
+
+    /* Ensure that no spurious interrupts are raised(do not change TMR1 and TMR4 IRQ status) */
+    irqSts = ZLL->IRQSTS;
+    irqSts &= ~(ZLL_IRQSTS_TMR1IRQ_MASK | ZLL_IRQSTS_TMR4IRQ_MASK);
+    irqSts |= ZLL_IRQSTS_TMR3MSK_MASK;
+    ZLL->IRQSTS = irqSts;
+
+    /* Unmask SEQ interrupt */
+    ZLL->PHY_CTRL &= ~ZLL_PHY_CTRL_SEQMSK_MASK;
+
+    /* start ED sequence */
+    ZLL->PHY_CTRL |= gCCA_c;
+}
+
+/*
+ * \brief Function is a call back for TX end interrupt.
+ *
+ * \param none
+ *
+ * \return none
+ */
+static void rf_handle_tx_end(bool framePending)
+{
+    /*Start receiver*/
+    rf_receive();
+
+    if (!device_driver.phy_tx_done_cb) {
+        return;
+    }
+
+    /*Call PHY TX Done API*/
+    if (need_ack) {
+        if (framePending) {
+            device_driver.phy_tx_done_cb(rf_radio_driver_id, rf_mac_handle, PHY_LINK_TX_DONE_PENDING, 1, 1);
+        } else {
+            // arm_net_phy_tx_done(rf_radio_driver_id, mac_tx_handle, PHY_LINK_TX_SUCCESS, 1, 1);
```

Comment:
can be removed

Reply:
I have updated the PR to remove this line.